### ISSUE TITLE
Blurry appearance when running on Wayland

### ISSUE THREAD

**Issue by spoorun** [ONLYOFFICE/DesktopEditors#561]
Running 6.1.0.90, on Manjaro KDE with Wayland
With fractional scaling at 2.25x on HiDPI screen

![Screenshot_20210201_114757](https://user-images.githubusercontent.com/627140/106455635-5f2a4b00-6484-11eb-9994-2a510bbb54ef.png)

**Comment by ShockwaveNN** [comment]
Hi there, currently out team have only HiDPI screen at our disposal, connected to Kubuntu 20.04
I've tried to run wayland here - but I got only black screen, and I don't think I have free time to figure out the reason of this

So could you try to setup non-wayland session on same pc+monitor combo to be sure that your problem is 100% in wayland

And on side note - could you descrbie your monitor resolution and DPI

**Comment by spoorun** [comment]
> Hi there, currently out team have only HiDPI screen at our disposal, connected to Kubuntu 20.04
> I've tried to run wayland here - but I got only black screen, and I don't think I have free time to figure out the reason of this
> 
> So could you try to setup non-wayland session on same pc+monitor combo to be sure that your problem is 100% in wayland
> 
> And on side note - could you descrbie your monitor resolution and DPI

Apologies for late reply..

My laptop monitor:
3840 x 2400 resolution
15.6" laptop screen
290 ppi

I can confirm that the issue occurs on another PC running Wayland and does not occur on my laptop or PC when not running Wayland. Exactly the same device and setup but without Wayland it's sharp and clear.

It may not be relevant, but I've had the same issue with blurry text when running Chrome and Chrome-based apps (and again, only on Wayland).

I worked-around it by launching them with:
Exec=google-chrome-beta --enable-features=UseOzonePlatform --ozone-platform=wayland

**Comment by ShockwaveNN** [comment]
> It may not be relevant, but I've had the same issue with blurry text when running Chrome and Chrome-based apps (and again, only on Wayland).

Hmm, this may be a case, since we're using chrome as base of our DesktopEditors, so the root problem may be on Chrome size, maybe already some known bug in chrome issue tracker

**Comment by spoorun** [comment]
> > It may not be relevant, but I've had the same issue with blurry text when running Chrome and Chrome-based apps (and again, only on Wayland).
> 
> Hmm, this may be a case, since we're using chrome as base of our DesktopEditors, so the root problem may be on Chrome size, maybe already some known bug in chrome issue tracker

So, Electron 12 is currently in beta and due for release next month, which will support Ozone Platform and Wayland. But whether it will require a commandline switch, as Chrome currently does is not currently obvious ( --enable-features=UseOzonePlatform --ozone-platform=wayland)

So I guess it's a waiting game for Chrome/Chromium/Electron support.

In the meantime, is there a way that HiDPI Wayland users like me can set that commandline switch to start DesktopEditors?

**Comment by ShockwaveNN** [comment]
> Electron 12 is currently in beta and due for release next month

Yeah, please note that we are not using Electron, we using `cef` directly

> In the meantime, is there a way that HiDPI Wayland users like me can set that commandline switch to start DesktopEditors?

Sorry I'm not aware of such and since I've got troubles running Wayland on my computer - currently I've no time to figure out it

**Comment by major-mayer** [comment]
I can confirm this issue also using Onlyoffice on Manjaro in combination with Wayland.

I know that native Wayland support has just landed on Chromium, so maybe it just takes a while until they also merge it into the CEF, but maybe it's already possible right now...
You can see the same behavior on other apps like Spotify which also use the CEF.
They are not using Wayland directly but instead rely on the presence of XWayland which is something like an X server within Wayland. 
And X does not have fractional scaling capabilities, that's where the blur comes from.

However, I am waiting hard on this feature, because it would make working with Onlyoffice on Linux systems which use Wayland as their display manager much more comfortable :smiley: 
Fedora for example is using Wayland by default for multiple years, even Ubuntu tried it out once IIRC.

BTW: In general I would say the Wayland support is better on Gnome in comparison to KDE, probably because they started implementing Wayland earlier.

**Comment by iMonZ** [comment]
Anything new on this issue?

**Comment by ShockwaveNN** [comment]
@iMonZ Sorry, I don't think so, mainly because part of issues related to CEF (chromium embedded framework) and update is quite tedious task for us

**Comment by spoorun** [comment]
Is this the snag point that needs fixing with CEF-Embedded?

https://bitbucket.org/chromiumembedded/cef/issues/2804/ozone-wayland-x11-support

**Comment by ShockwaveNN** [comment]
> needs fixing with CEF-Embedded?

I'm not 100% this issue is fixed in current stable releaser of CEF, but we use not very actual version and there is a lot of hard work updating it so I truly cannot say

**Comment by WhyNotHugo** [comment]
This is partially related to #382.

- XWayland itself does not fully support scaling, so results will always be somewhat blurry (it's just an upscaled surface).
- Native wayland support has been discussed in #382, and it likely the best way to get it to render properly.

**Comment by parrenin** [comment]
I confirm this issue still exists in OO 7.0 under KDE/wayland 5.24.3.
The problem is that under KDE, xwayland does not support scaling.
The issue is reported here:
https://bugs.kde.org/show_bug.cgi?id=449993
And there is a merge request here:
https://invent.kde.org/plasma/kwin/-/merge_requests/1959
Another way for OO to fix this issue would be to release a wayland-enabled version.

**Comment by ShockwaveNN** [comment]
> would be to release a wayland-enabled version.

I have no proper talk with our developers still, but judging by this issue https://bitbucket.org/chromiumembedded/cef/issues/2804/ozone-wayland-x11-support

We simply cannot release wayland-enabled version until this issue fixed (not counting any more issues we can find after that fix)
Correct me if I wrong, but we definitely using CEF and not planning to change that in the future

**Comment by BirdInFire** [comment]
The problem it's they use Xwayland, so it's an X server linked to wayland it's why it's blurry, if they had supported wayland nativly it's will not pose the problem, libreoffice has the same problem, and when they have fixed to work on native wayland poof problem gone.

**Comment by major-mayer** [comment]
Yeah, but that's nothing really new at this point...

Libreoffice supports Wayland both with their Qt and Gtk based backend by the way.

**Comment by BirdInFire** [comment]
> Yeah, but that's nothing really new at this point...
> 
> Libreoffice supports Wayland both with their Qt and Gtk based backend by the way.

i know but since onlyoffice is clearly better to handle word file (break less file), i hope to see native soon on it.

**Comment by WhyNotHugo** [comment]
Since CEF seems to be the blocker, probably using something like [Tauri](https://tauri.studio/en/) could be a good approach, but that's a lot of work.

**Comment by arjpar** [comment]
Any update on this issue? According to [this issue](https://bitbucket.org/chromiumembedded/cef/issues/2804/ozone-wayland-x11-support) it seems like CEF can run on Wayland now with some extra work, and there are some missing features. Not sure how much that pertains to OnlyOffice though. Maybe someone more experienced can chime in.

**Comment by ShockwaveNN** [comment]
@arjpar Sorry, currently there is no updates on this issue on ONLYOFFICE side

**Comment by pavelkalinin1985** [comment]
So...

**Comment by spoorun** [comment]
A year later, and Wayland is largely adopted across the board in Linux distros - any updates? Thanks

**Comment by LuckySpecialist** [comment]
Bug 63841 was issued for this problem.

**Comment by Smartich0ke** [comment]
Has there been any progress on this? Issue 63841 does not seem to exist.

**Comment by PumpkinXD** [comment]
tried [Arch wiki](https://wiki.archlinux.org/title/Wayland#Electron)'s tricks and OO still not running on wayland native

uh... did they blocked something?

**Comment by parrenin** [comment]
Any news on this issue?
I was wondering if there is any plan to update CEF in OnlyOffice next major version.

**Comment by HakaishinShwet** [comment]
any updates on this issue ? @LuckySpecialist

**Comment by LuckySpecialist** [comment]
@HakaishinShwet, sorry, there is no movement towards fixing this problem yet.
It has a low priority for developers.

**Comment by parrenin** [comment]
It is unfortunate that this issue has a low priority.
QHD displays are more and more common on laptops, and the only way to deal with them comfortably is with fractional scaling.
But fractional scaling on Gnome leads to a blurry OnlyOffice.

**Comment by repsac-by** [comment]
If you use GNOME and Archlinux, you can install [mutter-xwayland-scaling](https://aur.archlinux.org/packages/mutter-xwayland-scaling) and configure it as described in the pinned post. This will resolve the issue immediately for all applications using XWayland, including ONLYOFFICE.

**Comment by parrenin** [comment]
@repsac-by, thanks for the hint.
Unfortunately, I do not use Arch Linux so I cannot use this package.
But there is good news, Gnome is considering some better Xwayland scaling:
https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3567
Maybe this will help us before OnlyOffice supports Wayland natively.

**Comment by Nvixbl** [comment]
Majority of the users have HiDPI displays now as they have been out for almost a decade, also more distros are starting to drop support for X11 and have Wayland only by default.

Those distros are used by many linux beginners who don't know what X11 and Wayland are nor will they try to find a workaround for this issue, their first impression after using the app and seeing its blurry UI that causes headaches will make them to never consider it again and stick with Libreoffice that supports native Wayland.

**Comment by Smartich0ke** [comment]
I have tried Gnome's new xwayland scaling feature. It's an experimental feature that can be enabled in Gnome 47. It's no longer blurry, but onlyoffice seems to use a slightly larger gui scale than normal. Other apps work really well with it though.

I really wish this would get fixed soon, as wayland is becoming default on more and more distros and no wayland support in 2024 is kinda worrying.

**Comment by Nvixbl** [comment]
Majority of the users have HiDPI displays as they have been out for almost a decade and most distros are starting to drop X11 and have Wayland only by default.

Those distros are used by many linux beginners who won't know what X11 and Wayland nor will they try to find a fix for this issue, their first impression of the app will be its blurry UI that causes headaches making them never consider this app again and switch to Libreoffice that has native Wayland support.

 I hope the developers re-asses the priority of this issue as it's making the app barely usable for many of us.

**Comment by parrenin** [comment]
Chromium has not switched to Wayland yet, so I think there is a long way before OnlyOffice, which uses CEF, will switch to Wayland.
For fractional scaling, the best at the moment is to use the experimental XWayland scaling in Gnome 47.
I tested it and it seems to work well.

**Comment by parrenin** [comment]
I found this 7-month old video about CEF and Wayland:
https://youtu.be/LguQl74iiUs?feature=shared
In summary, there is a long way before CEF supports Wayland.

**Comment by parrenin** [comment]
Another related issue:
https://github.com/chromiumembedded/cef/issues/2804

**Comment by taprobane99** [comment]
Just to mention I am running this on Gnome 48 Wayland on a HIDIPI monitor (218 ppi) - everything is perfectly scaled and sharp at 200%

Default Gnome Setttings: sharp
Turning on experimental fractional scale buffer: blurry
Turning on experimental fractional scale buffer + let apps scale themselves: sharp

**Comment by ibrahimbayyinah** [comment]
Any solution so far for this? I'm running Hyprland on Arch Linux. The blur is quite annoying.

**Comment by gAlleb** [comment]
@ibrahimbayyinah

Try this in hyprland conf and set scaling to auto in OnlyOffice.

```
xwayland {
  force_zero_scaling = true
}
```

p.s. however i have another problem only with OnlyOffice on hyprland. Whenever you open OnlyOffice and then close it - hyprland would iidle at increased CPU (8-13%) until session restart or killing and starting over Xwayland process.

**Comment by WhyNotHugo** [comment]
On Thu, 3 Jul 2025, at 15:14, Felipe Avelar wrote:
> *FelipeFMA* left a comment (ONLYOFFICE/DesktopEditors#561) <https://github.com/ONLYOFFICE/DesktopEditors/issues/561#issuecomment-3032225021>
> can't yall just ditch xwayland? it makes everything laggy and weird...
> 

No, we can’t. OnlyOffice won’t work without it. That’s what this issue is all about.

**Comment by parrenin** [comment]
> i don't mean your guys, i mean the onlyoffice developers, why can't they just make onlyoffice wayland-native? it's the default for quite a big while, some distros are even dropping support for x11

Long story short: OnlyOffice Desktop Editors uses CEF (Chromium Embedded Framework) which does not support Wayland by default yet. So it is not in the hands of OnlyOffice developers. I expect it to happen in a year or two, once Chromium support Wayland by default.

**Comment by viggy96** [comment]
Chromium already has flags to use Wayland. 

Like "--ozone-platform-hint=auto". 

I've added flags like these to many electron apps to get them to act correctly in Wayland.

**Comment by parrenin** [comment]
Chromium 140 now defaults to Wayland on Linux.
CEF 140 will follow shortly.
So now this issue is in the hand of the OnlyOffice developers.

**Comment by martinoc000** [comment]
Not expected to see them working on it. Hope to be wrong.

**Comment by basicuser01** [comment]
I switched from OnlyOffice back to LibreOffice and Google Docs because of this issue.

**Comment by lastattemptdarling** [comment]
The sad part is that, they don't even bother answering 

I'm seriously thinking about flooding the repo with issues about wayland support

**Comment by WhyNotHugo** [comment]
On Sun, 26 Oct 2025, at 19:46, lastattemptdarling wrote:
> *lastattemptdarling* left a comment (ONLYOFFICE/DesktopEditors#561) <https://github.com/ONLYOFFICE/DesktopEditors/issues/561#issuecomment-3448768561>
> I'm seriously thinking about flooding the repo with issues about wayland support
> 

Don’t. Nothing good will come of doing so.

**Comment by parrenin** [comment]
> The sad part is that, they don't even bother answering
> I'm seriously thinking about flooding the repo with issues about wayland support

CEF 140 with support for Wayland is just out for a few weeks.
So there is really nothing wrong with OnlyOffice not supporting native Wayland yet at this stage.
Let us see if this comes out in 2026.

**Comment by Milli189** [comment]
yes please! Make onlyoffice run natively on Wayland!

**Comment by kleenleen** [comment]
Seriously, what are you waiting for to work on that issue ??

It's really late for completing this ticket.

**Comment by Milli189** [comment]
> Seriously, what are you waiting for to work on that issue ??
> 
> It's really late for completing this ticket.

I'm not sure this is the best attitude to have in this, as at the end of the day this is a free and open source project, I don't think pretending features if you don't pay the developers is the best way to ask them to do this.

I'm sure they will tackle this and give Wayland support eventually, as this is highly requested.

**Comment by kleenleen** [comment]
I agree, it was a tad too aggressive, but they are also selling their products to customers.
It's not a community project only.

**Comment by parrenin** [comment]
According to this ticket https://github.com/chromiumembedded/cef/issues/2804#issuecomment-3822946710 , the wayland support in CEF is not complete yet, missing in particular the merge of ANGLE for 3D accelerated rendering.
I suppose the OO developers are probably waiting for this.

**Comment by viggy96** [comment]
Would be nice if they confirmed that themselves

**Comment by thepythonguy-sm** [comment]
Instead of trying to make the OfficeSuite more usable, the devs are focusing on adding more AI slop, likely all vibe-coded too.

**Comment by martinoc000** [comment]
EuroOffice will be the way. People have enough of all this.

Too late to learn from mistakes.

**Comment by rusty-snake** [comment]
When it gets better desktop/mobile apps, yes. But at the moment there is nothing good.

**Comment by thepythonguy-sm** [comment]
> EuroOffice will be the way. People have enough of all this.
> 
> Too late to learn from mistakes.

Hopefully EuroOffice adopts wayland and GTK4 and we can finally have GPU acceleration. I seriously believe the "Use graphics acceleration when available" is probably a lie. CPU spikes 4x more than my GPU

**Comment by thepythonguy-sm** [comment]
> When it gets better desktop/mobile apps, yes. But at the moment there is nothing good.

As a mobile editor WPS Office is miles better than OnlyOffice, OO is not open source under mobile.

**Comment by rusty-snake** [comment]
> OO is not open source under mobile.

WPS not at all.

(we're getting OT)

**Comment by taprobane99** [comment]
> > EuroOffice will be the way. People have enough of all this.
> > Too late to learn from mistakes.
> 
> Hopefully EuroOffice adopts wayland and GTK4 and we can finally have GPU acceleration. I seriously believe the "Use graphics acceleration when available" is probably a lie. CPU spikes 4x more than my GPU

What does this even mean? The reason that native Wayland wasn't possible is that it wasn't merged into CEF until recently. Also, why would OnlyOffice adopt GTK4. They would have to develop a completely separate app to all their other platforms.

**Comment by thepythonguy-sm** [comment]
> > > EuroOffice will be the way. People have enough of all this.
> > > Too late to learn from mistakes.
> > 
> > 
> > Hopefully EuroOffice adopts wayland and GTK4 and we can finally have GPU acceleration. I seriously believe the "Use graphics acceleration when available" is probably a lie. CPU spikes 4x more than my GPU
> 
> What does this even mean? The reason that native Wayland wasn't possible is that it wasn't merged into CEF until recently. Also, why would OnlyOffice adopt GTK4. They would have to develop a completely separate app to all their other platforms.

Migrate to GTK4 to support 2026 software and hardware, not GTK3, where all the dialogs on Fedora Rawhide Linux are totally broken.
That might be true, but they could have built their own custom version of CEF/Electron. Besides now that ANGLE has been merged into CEF, the OO team will throw out a good reason why not to migrate over to Wayland.
OnlyOffice uses old GTK3 and Qt5, as far as I'm concerned, this could have been switched to in a matter of a year, no problem. Pretty much all is rendered through Chromium, I mean, KDE migrated to Qt6 in what two years? Over 500 projects, from the 1251,92 MiB install size, Qt5 code is like nothing. If you don't allow community members to contribute, then they better be good at implementing it themselves, which isn't the case as they're busy vibecoding garbage AI tools into OO.

**Comment by markspercyspencer** [comment]
> Majority of the users have HiDPI displays as they have been out for almost a decade and most distros are starting to drop X11 and have Wayland only by default.
> 
> Those distros are used by many linux beginners who won't know what X11 and Wayland nor will they try to find a fix for this issue, their first impression of the app will be its blurry UI that causes headaches making them never consider this app again and switch to Libreoffice that has native Wayland support.
> 
> I hope the developers re-asses the priority of this issue as it's making the app barely usable for many of us.

As a new Linux user I can but only confirm that this is what's happening. I tried OnlyOffice and as much as I appreciate the effort to provide for a good office suite, the blurry text is literally unacceptable in an app where TEXT and IMAGE are at the centre of everything. 

People spent literal hours each day on this app, writing/reading blurry text and interacting with a blurry interface. 

Please DO something! Wayland and HDPI screens are largely supported now.
